src/main/java/ch/uzh/ifi/hase/soprafs22/service/LobbyService.java

```diff
@@ -88,7 +88,6 @@ public ILobby createLobby(String token, String lobbyName, Visibility visibility,
         try {
             newLobby = lobbyManager.createLobby(lobbyName, visibility);
         }catch(SmallestIdNotCreatable e){
-           e.printStackTrace();
            String errorMessage = "The server could not generate a unique id";
            throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR, errorMessage);
```

Comment:
throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR, errorMessage, e);

If we add the cause "e" to the constructor we can trace the exception back to the origin.

Reply:
Good point, but the question is how does the json return object handle this. It is possible that we then leak implementation details through the API. I will quickly test what the API returns, or do you know it by chance ?

Reply:
_id\"; nested exception is ch.uzh.ifi.hase.soprafs22.exceptions.SmallestIdNotCreatable\r\n\tat_

_Caused by: ch.uzh.ifi.hase.soprafs22.exceptions.SmallestIdNotCreatable\r\n\tat ch.uzh.ifi.hase.soprafs22.lobby.LobbyManager.createLobby(LobbyManager.java:64)\r\n\tat ch.uzh.ifi.hase.soprafs22.service.LobbyService.createLobby(LobbyService.java:89)\r\n\t... 63 more\r\n",_

These two things get added to the trace field in the return json object, I don't think that we need to hide them and we get better traceability from it as you said. 

Thus I would add your change if you agree

Reply:
Agree at first sight.
For the sake of naming, do you want to add an Exception at SmallestIdNotCreatable so that it becomes SmallestIdNotCreatableException?

Reply:
Yea, that sounds better. I will definitely do that.